marklogic-client-api/src/main/java/com/marklogic/client/impl/RESTServices.java

```diff
@@ -450,37 +464,116 @@ public SingleAtomicCallField(String paramName, String paramValue) {
+	}
   }
-  static public class SingleNodeCallField extends CallField {
+  static public class UnbufferedSingleNodeCallField extends CallField implements SingleNodeCallField {
     private AbstractWriteHandle paramValue;
-    public SingleNodeCallField(String paramName, AbstractWriteHandle paramValue) {
+    public UnbufferedSingleNodeCallField(String paramName, AbstractWriteHandle paramValue) {
       super(paramName);
       this.paramValue = paramValue;
     }
+    @Override
     public AbstractWriteHandle getParamValue() {
       return paramValue;
     }
+    @Override
+    public BufferedSingleNodeCallField toBuffered() {
+    	if(!(paramValue instanceof BufferableHandle))
+    		throw new IllegalArgumentException("Default parameter must implement BufferableHandle.");
+    	BytesHandle bytesHandle = new BytesHandle((BufferableHandle) paramValue);
+    	return new BufferedSingleNodeCallField(super.paramName, Stream.of(bytesHandle));
+    }
+
   }
-  static public class MultipleNodeCallField extends CallField {
+  static public class UnbufferedMultipleNodeCallField extends CallField implements MultipleNodeCallField {
     private Stream<? extends AbstractWriteHandle> paramValues;
 
-    public MultipleNodeCallField(String paramName, Stream<? extends AbstractWriteHandle> paramValues) {
+    public UnbufferedMultipleNodeCallField(String paramName, Stream<? extends AbstractWriteHandle> paramValues) {
       super(paramName);
       this.paramValues = paramValues;
     }
+    
+    @Override
     public Stream<? extends AbstractWriteHandle> getParamValues() {
       return paramValues;
     }
+    
+    @Override
+    public BufferedMultipleNodeCallField toBuffered() {
+    	if(!(paramValues instanceof BufferableHandle))
+    		throw new IllegalArgumentException("Default parameter must implement BufferableHandle.");
+    	BytesHandle bytesHandle = new BytesHandle((BufferableHandle) paramValues);
+    	return new BufferedMultipleNodeCallField(super.paramName, Stream.of(bytesHandle));
+    }
+    
+  }
+  static public class BufferedMultipleAtomicCallField extends CallField implements MultipleAtomicCallField {
+	  
+	private String[] paramValues;
+	public BufferedMultipleAtomicCallField(String paramName, Stream<String> paramValues) {
+		super(paramName);
+        this.paramValues = (String[]) paramValues.toArray();
+    }
+  
+	@Override
+	public Stream<String> getParamValues() {
+		return Stream.of(paramValues);
+	}
+  }
+	
+  static public class BufferedMultipleNodeCallField extends CallField implements MultipleNodeCallField {
+	  
+		private Stream<BytesHandle> paramValues;
+		public BufferedMultipleNodeCallField(String paramName, Stream<BytesHandle> paramValues) {
+			super(paramName);
+	        this.paramValues = paramValues;
+	    }
+	  
+		@Override
+		public Stream<BytesHandle> getParamValues() {
+			return paramValues;
+		}
+	  
+  }
+  
+  static public class BufferedSingleNodeCallField extends CallField implements SingleNodeCallField {
```

Comment:
A single node call field takes a single node value.  So

- the constructor should take a single BytesHandle instead of a stream
- the private field should store a single BytesHandle
- the single value getter should return the private stream instead of null 
- there should not be a stream getter

Adding @Override marker annotations can help assure that the class is implementing the contract of SingleNodeCallField